src/magic_folder/config.py

```diff
@@ -620,6 +724,56 @@ def rotate_api_token(self):
             f.write(self._api_token)
         return self._api_token
 
+
+@attr.s
+@implementer(ITokenProvider)
+class MemoryTokenProvider(object):
+    """
+    Keep an in-memory token.
+    """
+    _api_token = attr.ib(default=None)
+
+    def get(self):
+        """
+        Retrieve the current token.
+        """
+        if self._api_token is None:
+            self.rotate()
```

Comment:
What does the other implementation do in this case?  My guess is it raises OSError.  Maybe it should auto-rotate too?